core/genesis.go

```diff
@@ -256,14 +260,20 @@ func (g *Genesis) configOrDefault(ghash common.Hash) *params.ChainConfig {
 }
 
 // ApplyOvmStateToState applies the initial OVM state to a state object.
-func ApplyOvmStateToState(statedb *state.StateDB) {
+func ApplyOvmStateToState(statedb *state.StateDB, xDomainMessengerAddress common.Address) {
 	for _, account := range vm.OvmStateDump.Accounts {
 		statedb.SetCode(account.Address, common.FromHex(account.Code))
 		statedb.SetNonce(account.Address, account.Nonce)
 		for key, val := range account.Storage {
 			statedb.SetState(account.Address, key, common.HexToHash(val))
 		}
 	}
+	// Set the storage slot associated with the cross domain messenger
+	// to the cross domain messenger address.
+	AddressManager := vm.OvmStateDump.Accounts["Lib_AddressManager"]
+	slot := common.HexToHash("0x515216935740e67dfdda5cf8e248ea32b3277787818ab59153061ac875c9385e")
+	value := common.BytesToHash(xDomainMessengerAddress.Bytes())
+	statedb.SetState(AddressManager.Address, slot, value)
```

Comment:
Can we also add a config flag for the Address Manager Owner address? (it's set at the 0x00...00 storage slot)

```suggestion
	slot := common.HexToHash("0x515216935740e67dfdda5cf8e248ea32b3277787818ab59153061ac875c9385e")
	value := common.BytesToHash(xDomainMessengerAddress.Bytes())
	statedb.SetState(AddressManager.Address, slot, value)
	slot := common.HexToHash("0x0000000000000000000000000000000000000000000000000000000000000000")
	value := common.BytesToHash(addressManagerOwner.Bytes())
	statedb.SetState(AddressManager.Address, slot, value)
```

Reply:
Added this functionality